index/src/executors/dex_executors/acala_dot_swap.rs

```diff
@@ -0,0 +1,143 @@
+use pink_subrpc::{create_transaction, send_transaction, ExtraParam};
+
+use crate::prelude::DexExecutor;
+use crate::traits::common::Error;
+use alloc::{
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
+use scale::Decode;
+use scale::{Compact, Encode};
+
+#[derive(Clone)]
+pub struct AcalaDotSwapExecutor {
+    rpc: String,
+}
+
+#[allow(dead_code)]
+impl AcalaDotSwapExecutor {
+    pub fn new(rpc: &str) -> Self
+    where
+        Self: Sized,
+    {
+        Self {
+            rpc: rpc.to_string(),
+        }
+    }
+}
+
+impl AcalaDotSwapExecutor {
+    fn aggregated_swap(
+        &self,
+        signer: [u8; 32],
+        path: Vec<u8>,
+        spend: u128,
+    ) -> core::result::Result<Vec<u8>, Error> {
+        let amount_out = Compact(1_u8);
```

Comment:
Why do we set a concrete out amount here?

Reply:
I have no idea how you manage to call the APIs with the amount being 0, does this code actually wor?: https://github.com/Phala-Network/phast-solver/blob/c3bcd40e53432bb1aae3fa7f2ca85317517cdbe0/src/dex_executor.ts#L95,